src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -205,6 +230,28 @@ async Task UpdateUnderLockAsync(UnconfiguredProject project, Action<ProjectRootE
             }
         }
 
+        public virtual async Task<IEnumerable<KeyValuePair<string, IEnumerable<string>>>> GetProjectConfigurationDimensionsAsync(Project project)
+        {
+            Requires.NotNull(project, nameof(project));
+
+            ImmutableArray<string> values = GetOrderedPropertyValues(project);
+            if (values.IsEmpty)
+            {
+                return ImmutableArray<KeyValuePair<string, IEnumerable<string>>>.Empty;
+            }
+            else
+            {
+                var dimensionValues = PooledArray<KeyValuePair<string, IEnumerable<string>>>.GetInstance();
+                dimensionValues.Add(new KeyValuePair<string, IEnumerable<string>>(DimensionName, values));
+                return dimensionValues.ToImmutableAndFree();
```

Comment:
Similarly this (and in other methods here) can just return an array, to simplify things and reduce allocations a little:

```suggestion
                return new[] { new KeyValuePair<string, IEnumerable<string>>(DimensionName, values) };
```